### PR TITLE
Support installation into /usr/local/include

### PR DESCRIPTION
**Pull request by jcwenger**
Recommend: cmake  -DCMAKE_INSTALL_PREFIX="/usr/local"

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,6 @@
+*.proto		text
```
Comment (rsjudka) [inline .gitattributes]: spaces vs tabs for some of these?

File: aasdk_proto/AVChannelSetupStatusEnum.proto [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+syntax="proto2";
+
+package f1x.aasdk.proto.enums;
+
+message AVChannelSetupStatus
+{
+    enum Enum
+    {
+        NONE = 0;
+		FAIL = 1;
```
Comment (rsjudka) [inline aasdk_proto/AVChannelSetupStatusEnum.proto]: tabs vs spaces on these?

File: aasdk_proto/ShutdownReasonEnum.proto [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
+
+syntax="proto2";
+
+package f1x.aasdk.proto.enums;
+
+message ShutdownReason
+{
+    enum Enum
+    {
+		NONE = 0;
```
Comment (rsjudka) [inline aasdk_proto/ShutdownReasonEnum.proto]: indentation seems off

File: aasdk_proto/VideoFocusModeEnum.proto [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
+
+syntax="proto2";
+
+package f1x.aasdk.proto.enums;
+
+message VideoFocusMode
+{
+    enum Enum
+    {
+		NONE = 0;
```
Comment (rsjudka) [inline aasdk_proto/VideoFocusModeEnum.proto]: indentation again?

File: aasdk_proto/VideoFocusReasonEnum.proto [hunk 5]
```diff
@@ -1,31 +1,31 @@
...
+
+syntax="proto2";
+
+package f1x.aasdk.proto.enums;
+
+message VideoFocusReason
+{
+    enum Enum
+    {
+		NONE = 0;
```
Comment (rsjudka) [inline aasdk_proto/VideoFocusReasonEnum.proto]: indentation seems off

File: include/f1x/aasdk/IO/Promise.hpp [hunk 6]
```diff
@@ -20,6 +20,7 @@
 
 #include <functional>
 #include <boost/asio.hpp>
+#include <boost/core/noncopyable.hpp>
```
Comment (rsjudka) [inline include/f1x/aasdk/IO/Promise.hpp:23]: thanks for adding this im assuming you also upgraded boost? :p

File: src/Channel/ServiceChannel.cpp [hunk 7]
```diff
@@ -38,7 +38,12 @@ ServiceChannel::ServiceChannel(boost::asio::io_service::strand& strand,
 
 void ServiceChannel::send(messenger::Message::Pointer message, SendPromise::Pointer promise)
 {
+#if BOOST_VERSION < 106600
```
Comment (rsjudka) [inline src/Channel/ServiceChannel.cpp:41]: so `context` doesnt exist in older boost? would like to minimize any conditionals like this

**Comment by rsjudka** [comment]
what happened with all the proto stuff? did line ending change or something?

File: .gitattributes (hunk 1 above)
Comment (jcwenger) [reply .gitattributes]: gitattributes won't deal with indentation, only line endings.  you'd need a pre-commit hook against some user tool

File: include/f1x/aasdk/IO/Promise.hpp (hunk 6 above)
Comment (jcwenger) [reply include/f1x/aasdk/IO/Promise.hpp:23]: noncopyable is used there, so the header is needed.  Boost from buster somehow includes it implicitly.  buildroot pulls a later version of boost (1.72) -- and that version, the header isn't implicit and so its needed here.

File: src/Channel/ServiceChannel.cpp (hunk 7 above)
Comment (jcwenger) [reply src/Channel/ServiceChannel.cpp:41]: context got just got renamed.  Yaay for API stability I guess.  The later boost used by buildroot won't work without the conditional.  boost 1.66 is the first appearance of the new name and the deprecation of the existing name.

File: aasdk_proto/AVChannelSetupStatusEnum.proto (hunk 2 above)
Comment (jcwenger) [reply aasdk_proto/AVChannelSetupStatusEnum.proto]: tabs versus spaces across _all_ these files are untouched -- all I did was fix line endings to be native.

File: src/Channel/ServiceChannel.cpp (hunk 7 above)
Comment (rsjudka) [reply src/Channel/ServiceChannel.cpp:41]: ugh not a fan but i guess what can you do

**Comment by rsjudka** [comment]
lgtm, feel free to merge

**Pull request merged**